core/modules/widgets/reveal.js

```diff
@@ -183,18 +193,18 @@ RevealWidget.prototype.compareStateText = function(state) {
 };
 
 RevealWidget.prototype.readPopupState = function(state) {
-	var popupLocationRegExp = /^\((-?[0-9\.E]+),(-?[0-9\.E]+),(-?[0-9\.E]+),(-?[0-9\.E]+)\)$/,
-		match = popupLocationRegExp.exec(state);
+	var match = $tw.utils.Popup.popupLocationRegExp.exec(state);
```

Comment:
While I think the regexp definition should be done only once. I'm not so happy with the implementation. 

`$tw.utils.Popup.popupLocationRegExp.exec()` defines the "matching groups" and returns a match. 
but
the `if (match)` ... `match[1-2-3]` are done here ... So there is a hardcoded dependency between 2 files now. ... 

IMO the `$tw.utils.Popup` should also return an object that contains the `absolute, left, top ... ` values. So if changes to the group matches need to be made, they can be done in 1 file only. 

reveal.js and action-popup.js should call the functions and get the results they need, without the `parseFloat(match[x])` code

Reply:
You are right. My approach was incomplete. I've reworked the commit and put a `parseCoordinates` method into `popup.js`.